src/tmpfiles/tmpfiles.c

```diff
@@ -2003,6 +2005,177 @@ static int glob_item_recursively(Item *i, fdaction_t action) {
         return r;
 }
 
+static int rm_if_wrong_type_safe(
+                mode_t mode,
+                int parent_fd,
+                const struct stat *parent_st /* Only used if follow is true. */,
+                const char *name,
+                bool follow) {
+        _cleanup_free_ char *parent_name = NULL;
+        struct stat st;
+        int r;
+        const char *e;
+
+        assert(name);
+        assert((mode & ~S_IFMT) == 0);
+        if (follow)
+                assert(parent_st);
+
+        e = strrchr(name, '/');
+        if (e)
+                return -EINVAL;
```

Comment:
filename_is_valid()?

Reply:
Our functions should either log about all errors or about none, see https://systemd.io/CODING_STYLE/

THis function logs about some, but here it doesn't. Figure out what kind of function this should be, and stick to it, to avoi duplicate or missing error logging